sdk/storage/azure-storage-common/src/storage_credential.cpp

```diff
@@ -96,10 +96,18 @@ namespace Azure { namespace Storage { namespace _internal {
     std::string sas = getWithDefault(connectionStringMap, "SharedAccessSignature");
     if (!sas.empty())
     {
-      connectionStringParts.BlobServiceUrl.AppendQueryParameters(sas);
-      connectionStringParts.DataLakeServiceUrl.AppendQueryParameters(sas);
-      connectionStringParts.FileServiceUrl.AppendQueryParameters(sas);
-      connectionStringParts.QueueServiceUrl.AppendQueryParameters(sas);
+      if (sas[0] != '?')
+      {
+        sas = '?' + sas;
+      }
```

Comment:
When would this condition be true?

Reply:
![image](https://user-images.githubusercontent.com/52266153/113653047-9eb12580-96c7-11eb-8704-ee797eff080d.png)


![image](https://user-images.githubusercontent.com/52266153/113653212-fa7bae80-96c7-11eb-9b08-cfd291b75abd.png)

sometimes the SAS token has a leading `?` while sometimes it doesn't

Reply:
Why does it not impact correctness for us to add back a `?` to the generated URL when the SAS token doesn't start with a leading `?`?

Reply:
There has to be a `?` between path and query parameters in URL. So if there's no `?` in query parameters, we add one. If there's already one, we don't add.